plugin/src/main/groovy/com/novoda/buildproperties/internal/ExceptionFactory.groovy

```diff
@@ -0,0 +1,10 @@
+package com.novoda.buildproperties.internal
+
+interface ExceptionFactory {
+
+    void setAdditionalMessage(String additionalMessage)
```

Comment:
I know, it looks weird. But unfortunately a properties set is mutable, so this was way to inject the description to the error message.